cypress-io/cypress
# feat: Create spec from Vue component

- Closes https://github.com/cypress-io/cypress/issues/22836 

### User facing changelog


For Vue projects in component testing, choose to generate a spec file from a list of the components that exist in your project

### Additional details


This PR makes it so that when a user opens a Vue project for component testing and click the "+ New Spec" button, they are presented with a list of the components in their project. They can click on a component and a component spec will be created with the name of the component and placed in the directory where the component exists.

The test will be scaffolded to match the syntax of either a Vue 2 or Vue 3 app depending on which version of the framework the project is using.

### Steps to test


1. Create a new Vue (2 or 3) project that has some components
2. Configure the project for component testing
3. Open component testing in a browser
4. Click the "+ New Spec" button in the top-right of the spec list
5. Confirm that you are presented with a list of the components in your application
6. Click a component and confirm that a test is created with the correct name and template
7. Run the generated spec (it should pass unless the component has some required props that we're not passing)
8. Verify that creating empty specs still works for React projects and e2e projects

### How has the user experience changed?


Waiting to post screenshots until we get design feedback.

### Design Questions

1. Should users be able to create from component **as well as** create empty specs in Vue projects, now that create from component is available in these projects?
2. If there is only one option, should we show the card for the option in the modal, or should we immediately go to the first step (this is what we do now, in this branch, for "create from component").
3. What should the create from component card look like (see linked issue for a screenshot of the existing one)? Do we need to make changes to the existing one? If the answer to # 1 is yes, then I think we should at least update the icon so that it looks different from the "create empty" card.

### PR Tasks


- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
- [ ] Design approval
